api/src/main/java/org/openmrs/module/fhir2/api/search/SearchQueryInclude.java

```diff
@@ -278,13 +286,23 @@ private IBundleProvider handleEncounterReverseInclude(ReferenceAndListParam para
 	private IBundleProvider handleMedicationReverseInclude(ReferenceAndListParam params, String targetType) {
 		switch (targetType) {
 			case FhirConstants.MEDICATION_REQUEST:
-				return medicationRequestService.searchForMedicationRequests(null, null, null, null, params, null, null,
+				return medicationRequestService.searchForMedicationRequests(null, null, null, null, params, null, null, null,
 				    null);
 		}
 		
 		return null;
 	}
 	
+	private IBundleProvider handlePrescriptionReverseInclude(ReferenceAndListParam params, String targetType) {
+		switch (targetType) {
+			case FhirConstants.MEDICATION_DISPENSE:
+				MedicationDispenseSearchParams medicationDispenseSearchParams = new MedicationDispenseSearchParams();
+				medicationDispenseSearchParams.setMedicationRequest(params);
+				return medicationDispenseService.searchMedicationDispenses(medicationDispenseSearchParams);
```

Comment:
Presumably this would be the place to decide if MedicationDispense is supported or not and to do the right thing?  Is the service not capable of doing that?

Reply:
So you were correct @mseaton this *was* returning an empty bundle if MedicationDispense is not supported.... I modified line 157 above to check if the provider is empty and this fixes the issue... now if you request a MedicationRequest with a MedicationDispense revinclude against a pre 2.6.x core, it doesn't crash but simply doesn't return any MedicationDispense events.

Reply:
Yeah, this is my interpretation of how the FHIR API is meant to be implemented, i.e., in general we should accept as many requests as possible and just ignore parts of the request we can't process.